src/SpineAnimation.ts

```diff
@@ -37,79 +41,97 @@ export class SpineAnimation extends Script {
   }
 
   @ignoreClone
-  private _skeletonData: SkeletonData;
+  setting: SpineRenderSetting;
+  @ignoreClone
+  skeletonData: SkeletonData;
   @ignoreClone
-  private _skeleton: Skeleton;
+  initialSkinName: string;
+  /* @internal */
   @ignoreClone
-  private _state: AnimationState;
+  _primitive: Primitive;
+  /* @internal */
   @ignoreClone
-  private _tempOffset: Vector2 = new Vector2();
+  _subPrimitives: SubPrimitive[];
   @ignoreClone
-  private _tempSize: Vector2 = new Vector2();
+  protected _skeleton: Skeleton;
   @ignoreClone
-  private _tempArray: Array<number> = [0, 0];
+  protected _state: AnimationState;
   @ignoreClone
-  protected _meshGenerator: MeshGenerator;
+  protected _spineGenerator: SpineGenerator;
   @ignoreClone
-  setting: SpineRenderSetting;
+  private _animationName: string;
+  @ignoreClone
+  private _loop: boolean = false;
 
-  autoUpdateBounds: boolean = false;
-  noPause: boolean = false;
+  get animationName() {
+    return this._animationName;
+  }
 
-  get skeletonData() {
-    return this._skeletonData;
+  set animationName(value: string) {
+    this._animationName = value;
+    if (this.state) {
+      if (value) {
+        this.state.setAnimation(0, value, this.loop);
+      } else {
+        this.state.setEmptyAnimation(0, 0);
+      }
+    }
   }
 
-  get skeleton() {
-    return this._skeleton;
+  get loop() {
+    return this._loop;
   }
 
-  get state() {
-    return this._state;
+  set loop(value: boolean) {
+    const entry = this.state && this.state.getCurrent(0);
+    this._loop = value;
+    if (entry) {
+      entry.loop = value;
+    }
   }
 
-  get mesh() {
-    return this._meshGenerator.mesh;
+  get state() {
+    return this._state;
   }
 
-  set scale(v: number) {
-    if (this._skeleton) {
-      this._skeleton.scaleX = v;
-      this._skeleton.scaleY = v;
-    }
+  get skeleton() {
+    return this._skeleton;
   }
 
   constructor(entity: Entity) {
     super(entity);
-    this._meshGenerator = new MeshGenerator(this.engine, entity);
+    this._spineGenerator = new SpineGenerator(this.engine, this);
```

Comment:
Whether the same `skeletonData` can share an `SpineGenerator`?

Reply:
![image](https://github.com/galacean/engine-spine/assets/7768919/60193477-54fe-452b-a88c-4faeed50a70b)